org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceDiagnosticsHandlerTest.java

```diff
@@ -101,6 +101,28 @@ public void testToDiagnosticsArray() throws Exception {
 
 	}
 
+	@Test
+	public void testTaskMarkers() throws Exception {
+		InitHandler initHandler = new InitHandler(projectsManager, preferenceManager, connection);
+		initHandler.addWorkspaceDiagnosticsHandler();
+		//import project
+		importProjects("eclipse/hello");
+
+		ArgumentCaptor<PublishDiagnosticsParams> captor = ArgumentCaptor.forClass(PublishDiagnosticsParams.class);
+		verify(connection, atLeastOnce()).publishDiagnostics(captor.capture());
+
+		List<PublishDiagnosticsParams> allCalls = captor.getAllValues();
+		Collections.reverse(allCalls);
+		projectsManager.setConnection(client);
+
+		Optional<PublishDiagnosticsParams> taskDiags = allCalls.stream().filter(p -> p.getUri().endsWith("TaskMarkerTest.java")).findFirst();
+		assertTrue("No TaskMarkerTest.java markers were found", taskDiags.isPresent());
+		List<Diagnostic> diags = taskDiags.get().getDiagnostics();
+		assertEquals("Some marker is missing", 3, diags.size());
```

Comment:
assertions should display current list of diagnostics, to help troubleshoot failures